<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>How not to be engulfed by MAGMA: instructions and model descriptions • magma.gsi</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How not to be engulfed by MAGMA: instructions and model descriptions">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">magma.gsi</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/age_prior_comparison.html">Age priors configurations comparison</a></li>
    <li><a class="dropdown-item" href="../articles/magma-vignette.html">How not to be engulfed by MAGMA: instructions and model descriptions</a></li>
    <li><a class="dropdown-item" href="../articles/mdl_comparison.html">Comparing MAGMA model algorithms</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boppingshoe/magma.gsi/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>How not to be engulfed by MAGMA: instructions and model descriptions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/boppingshoe/magma.gsi/blob/master/vignettes/magma-vignette.Rmd" class="external-link"><code>vignettes/magma-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>magma-vignette.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boppingshoe/magma.gsi" class="external-link">magma.gsi</a></span><span class="op">)</span></span>
<span><span class="co"># devtools::load_all()</span></span></code></pre></div>
<p>This document contains the background information for Mark and
Age-enhanced Genetic Mixture Analysis (MAGMA) and is organized in three
parts: overview descriptions of the MAGMA model, instructions on the
latest in the software programs to input data, run the MAGMA model, and
summarize results, and detailed descriptions of the MAGMA model and its
mathematical theory (the fun stuff).</p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>MAGMA is a Bayesian genetic stock identification model first
developed by the Gene Conservation Lab (GCL) biometrician Jim Jasper.
MAGMA is based on the Pella-Masuda model (Pella &amp; Masuda, 2001) and
incorporates information on age and hatchery group membership using
matched scales and otoliths to allow more detailed stock composition
estimates.</p>
<p>Mainly, MAGMA estimates two sets of parameters: population and age.
For each mixture (or a stratum), MAGMA estimates a vector of population
proportions of wild and hatchery groups. For all mixtures (or all
strata) within a year, MAGMA estimates a matrix of age proportions with
each row represents a wild or hatchery population. Populations are then
combined to represent a reporting group or a fishery stock. An
age-by-stock composition, information that is often required for run
reconstruction models, is simply the product of age and stock
parameters. In a single-stratum example shown below, we can see how the
age-by-stock composition is presented. A matrix of made-up age
proportions is as follows:</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Age 1</th>
<th align="left">Age 2</th>
<th align="left">Age 3</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Stock A</td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.2</mn><annotation encoding="application/x-tex">0.2</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.3</mn><annotation encoding="application/x-tex">0.3</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.5</mn><annotation encoding="application/x-tex">0.5</annotation></semantics></math></td>
</tr>
<tr class="even">
<td align="left">Stock B</td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.4</mn><annotation encoding="application/x-tex">0.4</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.5</mn><annotation encoding="application/x-tex">0.5</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.1</mn><annotation encoding="application/x-tex">0.1</annotation></semantics></math></td>
</tr>
</tbody>
</table>
<p>The made-up stock proportions are 0.7 and 0.3 for stock A and B in
this stratum. Multiplying the age and stock proportions for each
stock:</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Age 1</th>
<th align="left">Age 2</th>
<th align="left">Age 3</th>
<th align="left"></th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Stock A</td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.2</mn><annotation encoding="application/x-tex">0.2</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.3</mn><annotation encoding="application/x-tex">0.3</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.5</mn><annotation encoding="application/x-tex">0.5</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>×</mo><mn>0.7</mn></mrow><annotation encoding="application/x-tex">\times 0.7</annotation></semantics></math></td>
</tr>
<tr class="even">
<td align="left">Stock B</td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.4</mn><annotation encoding="application/x-tex">0.4</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.5</mn><annotation encoding="application/x-tex">0.5</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.1</mn><annotation encoding="application/x-tex">0.1</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>×</mo><mn>0.3</mn></mrow><annotation encoding="application/x-tex">\times 0.3</annotation></semantics></math></td>
</tr>
</tbody>
</table>
<p>The age-by-stock composition<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Currently, MAGMA output for age/stock comp is
standardized by each stock, so the age proportions within each stock add
up to 1.&lt;/p&gt;"><sup>1</sup></a> for this stratum is:</p>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Age 1</th>
<th align="left">Age 2</th>
<th align="left">Age 3</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Stock A</td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.14</mn><annotation encoding="application/x-tex">0.14</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.21</mn><annotation encoding="application/x-tex">0.21</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.35</mn><annotation encoding="application/x-tex">0.35</annotation></semantics></math></td>
</tr>
<tr class="even">
<td align="left">Stock B</td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.12</mn><annotation encoding="application/x-tex">0.12</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.15</mn><annotation encoding="application/x-tex">0.15</annotation></semantics></math></td>
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.03</mn><annotation encoding="application/x-tex">0.03</annotation></semantics></math></td>
</tr>
</tbody>
</table>
<p>After all age/stock compositions in other strata are calculated in
the same fashion, they are multiplied by the harvest proportions of
their corresponding strata and summed up to get a weighted average
age/stock composition for the whole fishery.</p>
<p>Age proportions are estimated by combining all mixtures/strata within
a year, which can be counter-intuitive at the first glance. After all,
it may not be reasonable to assume that all mixtures share the same age
distribution. However, in the MAGMA model, age compositions are adjusted
according to stock proportions for each mixture (i.e., stratum). Because
stock proportions are different from mixtures to mixtures, the
differences in stock compositions drive the differences in age
compositions between mixtures. In the previous example, stock A consists
mainly (50%) age 3 fish and is 70% of the total mixture population.
Therefore, age 3 fish from stock A is the dominant class in the mixture
population. In another stratum, stock B might be the majority of the
mixture populations. In which case, we would expect to see that age 2 to
be the dominant class because compositions are driven mostly by stock
B.</p>
<p>It is not required that all individuals in the data set have an
observed genotype or age. Base on data available, MAGMA model assigns a
most likely membership for population and/or age to each individual that
has unobserved genotype and/or age. However, there should be an adequate
sample of genotyped individuals to represent each stratum in the data
set. When there is an inadequate number (small or zero sample size) of
genotyped individuals in a stratum, MAGMA model estimates population
proportions of that stratum based on the overall proportions of all
strata combined.</p>
<p>Estimation of age and population compositions in the MAGMA model is
done through an algorithm called the Gibbs sampler. The process is
initialized by stochastically assigning all individuals with unobserved
identities a group membership for population or age class based on
specified priors, then the proportions for populations and age classes
are “estimated” by drawing values from the full conditional
distributions<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;The distribution of a parameter conditioned on the data
and other parameters in the model.&lt;/p&gt;"><sup>2</sup></a> of the population and age classes. The full
conditional distributions are mainly based on numbers of individuals
counted in each population and age group. The Gibbs sampler proceeds in
the following steps:</p>
<ol style="list-style-type: decimal">
<li>Determine population memberships of mixture individuals by:</li>
</ol>
<ul>
<li>stochastically assign a (wild) population membership to each
individual without observed identity but with genotypes based on its
genotypic frequencies and population proportions, and</li>
<li>stochastically assign a (wild) population membership to each
individual without observed identity and without genotypes based on
population proportions.</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li><p>If individual’s age is unobserved, assign an age based on the age
composition of its assigned population membership.</p></li>
<li><p>Draw updated values for population and age proportions and
baseline allele frequencies from their full conditional distributions
based on updates for:</p></li>
</ol>
<ul>
<li>tallies of individuals in each age class for each population
(assigned and observed) for both wild and hatchery groups and</li>
<li>specified prior values.</li>
</ul>
<p>This algorithm is repeated until simulations converged to the
posterior distribution of the parameters, usually it takes thousands of
iterations. We adapted a modified version of the conditional genetic
stock identification (conditional GSI; Moran &amp; Anderson 2018) model
for running MAGMA. This modification speeds up the Gibbs sampler
algorithm compared to the conventional “fully Bayesian” approach because
baseline allele frequencies are only updated every 10<sup>th</sup>
iteration in the modified conditional GSI algorithm. Details of the
fully Bayesian and conditional GSI models are covered in the Background
and methods section.</p>
<p>Using the posterior distribution, we summarize the point estimates
and credible intervals for the age/population compositions and model
convergence diagnostics in the output statistics.</p>
</div>
<div class="section level2">
<h2 id="how-to-use-magma">How to use MAGMA<a class="anchor" aria-label="anchor" href="#how-to-use-magma"></a>
</h2>
<div class="section level3">
<h3 id="data-format">Data Format<a class="anchor" aria-label="anchor" href="#data-format"></a>
</h3>
<p>Before running <code><a href="../reference/magmatize_data.html">magmatize_data()</a></code> function, we need to set
up a <em>data</em> folder in the working directory. The <em>data</em>
folder is where you put the files to compile input data for running the
MAGMA model. Those files are: <em>baseline.RData</em>,
<em>group_namesXXX.txt</em>, <em>groupsXXX.txt</em>,
<em>harvestXXX.txt</em>, <em>metadata.txt</em>, and
<em>mixture.RData</em>.</p>
<p><em>group_namesXXX.txt</em> and <em>groupsXXX.txt</em> are saved with
fishery extension. For example, <em>group_namesFAKE.txt</em> and
<em>groupsFAKE.txt</em> are files for the “FAKE” fishery.
<em>group_namesXXX.txt</em> contains the names of the reporting groups
in order of their group identifying numbers (groupvec).
<em>groupsXXX.txt</em> contains the groupvec for each population in the
data. The following shows the format of <em>group_namesXXX.txt</em> and
<em>groupsXXX.txt</em>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html" class="external-link">read_table</a></span><span class="op">(</span><span class="st">"data/group_namesFAKE.txt"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 1</span></span></span>
<span><span class="co">#&gt;   D1     </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Koyukuk</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Tanana </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> UpperUS</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Canada</span></span>
<span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html" class="external-link">read_table</a></span><span class="op">(</span><span class="st">"data/groupsFAKE.txt"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 43 × 2</span></span></span>
<span><span class="co">#&gt;    SOURCE                                 D1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> KHENS01                                 1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> KHENS07.KHENS15                         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> KSFKOY03                                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> KMFKOY10.KMFKOY11.KMFKOY12.KMFKOY13     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> KKANT05                                 2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> KCHAT01.KCHAT07                         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> KCHENA01                                2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> KSALC04.KSALC05                         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> KGOODP06.KGOODP07.KGOODP11.KGOODP12     2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> KBEAV97                                 3</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 33 more rows</span></span></span></code></pre></div>
<p><em>harvestXXX.txt</em> is also saved with a fishery extension.
Harvest file contains the number of harvest for each sampling week in
each district and subdistrict. The following shows an example for the
“FAKE” fishery harvest. Note that the column names in the harvest file
have to be in all capital letters.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html" class="external-link">read_table</a></span><span class="op">(</span><span class="st">"data/harvestFAKE.txt"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 5</span></span></span>
<span><span class="co">#&gt;    YEAR DISTRICT SUBDISTRICT STAT_WEEK HARVEST</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>  <span style="text-decoration: underline;">2</span>023        1           1         1    <span style="text-decoration: underline;">1</span>989</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span>  <span style="text-decoration: underline;">2</span>023        1           1         2    <span style="text-decoration: underline;">4</span>414</span></span></code></pre></div>
<p><em>metadata.txt</em> contains the information where (district and
subdistrict) and when (week) each fish in the data set was collected.
Each fish is identified by <code>SILLY_VIAL</code>, an unique
identifier. <em>metadata.txt</em> also contains information for age and
origin for each fish, if they are observed. Age is recorded in the
European system and without a “.” between the fresh and salt water ages.
Origin of the fish is recorded as “WILD” for the natural origin fish,
and a designated four letter code for each specific hatchery. The
following shows the format for <em>metadata.txt</em>. Note that the
column names have to be in all capital letters.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_table.html" class="external-link">read_table</a></span><span class="op">(</span><span class="st">"data/metadata.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 9</span></span></span>
<span><span class="co">#&gt;   SILLY_VIAL  YEAR STAT_WEEK DISTRICT SUBDISTRICT AGE_EUROPEAN SOURCE SILLY_CODE</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> FAKE_KBEA…  <span style="text-decoration: underline;">2</span>023         1        1           1           22 WILD   UpperUS   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> FAKE_KBEA…  <span style="text-decoration: underline;">2</span>023         2        1           1           32 WILD   UpperUS   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> FAKE_KBEA…  <span style="text-decoration: underline;">2</span>023         1        1           1           32 WILD   UpperUS   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> FAKE_KBEA…  <span style="text-decoration: underline;">2</span>023         2        1           1           22 WILD   UpperUS   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> FAKE_KBEA…  <span style="text-decoration: underline;">2</span>023         1        1           1           32 WILD   UpperUS   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> FAKE_KBEA…  <span style="text-decoration: underline;">2</span>023         1        1           1           <span style="color: #BB0000;">NA</span> WILD   UpperUS   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: real_age &lt;dbl&gt;</span></span></span></code></pre></div>
<p><em>baseline.RData</em> contains the genetic information for the
baseline populations in <code>.gcl</code> files. <em>baseline.RData</em>
is created by loading GCL baseline files onto the <em>R</em> environment
and run <code>save.image(file = "baseline.RData")</code>.</p>
<p><em>mixture.RData</em> contains the <code>.gcl</code> file(s) for the
mixture (samples with genetic information) and the fishery name (as a
character string). <em>mixture.RData</em> is also created by
<code><a href="https://rdrr.io/r/base/save.html" class="external-link">save.image()</a></code> in <em>R</em>. You have the option to include
the fishery name in <em>mixture.data</em> or specify in the
<code>magmatized_data()</code> function instead. If you specified at
both places with different fishery names, the one in the
<em>mixture.RData</em> will take precedence. This option allows users to
summarize MAGMA output under different reporting group setups without
rerunning the model again.</p>
</div>
<div class="section level3">
<h3 id="compile-input-files">Compile Input Files<a class="anchor" aria-label="anchor" href="#compile-input-files"></a>
</h3>
<p>To run MAGMA model, the input files need to be compiled first using
<code><a href="../reference/magmatize_data.html">magmatize_data()</a></code>. The function automatically formats the
input data as a list. Users will need to assign the list as an
<em>R</em> object as shown in the code below.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">yomamafat</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/magmatize_data.html">magmatize_data</a></span><span class="op">(</span>wd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                 age_classes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">13</span>, <span class="fl">21</span>, <span class="fl">22</span>, <span class="fl">23</span>, <span class="fl">31</span>, <span class="fl">32</span>, <span class="st">"other"</span><span class="op">)</span>,</span>
<span>                 fishery <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 loci_names <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                 save_data <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Compiling input data, may take a minute or two...</span></span>
<span><span class="co">#&gt; FAKE is the fishery identified in the mixture.RData</span></span>
<span><span class="co">#&gt; No missing hatcheries</span></span>
<span><span class="co">#&gt; Time difference of 9.823383 secs</span></span></code></pre></div>
<p>The function gives you the option to save the compiled input data.
The default is <code>save_data = TRUE</code>, and it will save the data
with the fishery extension as <em>magma_dataXXX.Rds</em> in the
<em>data</em> folder in your specified directory.</p>
<p>Age classes for the analysis is identified at this step. User can
specify the age classes or let MAGMA choose what age classes to
estimate. By default, MAGMA identifies the ranges for freshwater and
saltwater ages in metadata and expand the age classes using the age
ranges. For example, if the observed age classes are: 12, 13, 21, 22,
23, and 31, MAGMA would expand the age classes to 11, 12, 13, 21, 22,
23, 31, 32, and 33. If the analysis only has five major classes: 11, 12,
21, 22, and 31, user can specify an “other” group to include ages 13,
23, 32, and 33. In a similar fashion, user can specify a “0X” age to
catch all 0 freshwater ages.</p>
<p>MAGMA user can provide a list of loci as a check against baseline and
mixture data (optional). A warning message is returned if there are
difference between the data sets and the list provided.</p>
<p>The formatted input files have the following structure:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">yomamafat</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 16</span></span>
<span><span class="co">#&gt;  $ x           : int [1:149, 1:803] 2 1 2 2 2 1 1 0 1 2 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:149] "FAKE_KBEAV97_22" "FAKE_KBEAV97_31" "FAKE_KBEAV97_46" "FAKE_KBEAV97_60" ...</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:803] "GTH2B-550_1" "GTH2B-550_2" "NOD1_1" "NOD1_2" ...</span></span>
<span><span class="co">#&gt;  $ y           : int [1:43, 1:803] 144 205 62 54 140 72 236 204 156 137 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:43] "KHENS01" "KHENS07.KHENS15" "KSFKOY03" "KMFKOY10.KMFKOY11.KMFKOY12.KMFKOY13" ...</span></span>
<span><span class="co">#&gt;   .. ..$ : chr [1:803] "GTH2B-550_1" "GTH2B-550_2" "NOD1_1" "NOD1_2" ...</span></span>
<span><span class="co">#&gt;  $ metadat     :'data.frame':    149 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ district: int [1:149] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span><span class="co">#&gt;   ..$ subdist : int [1:149] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span><span class="co">#&gt;   ..$ week    : int [1:149] 1 2 1 2 1 1 1 1 1 1 ...</span></span>
<span><span class="co">#&gt;   ..$ iden    : int [1:149] NA NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;   ..$ age     : int [1:149] 5 8 8 5 8 NA NA 8 NA NA ...</span></span>
<span><span class="co">#&gt;  $ harvest     :'data.frame':    2 obs. of  5 variables:</span></span>
<span><span class="co">#&gt;   ..$ YEAR       : int [1:2] 2023 2023</span></span>
<span><span class="co">#&gt;   ..$ DISTRICT   : int [1:2] 1 1</span></span>
<span><span class="co">#&gt;   ..$ SUBDISTRICT: int [1:2] 1 1</span></span>
<span><span class="co">#&gt;   ..$ STAT_WEEK  : int [1:2] 1 2</span></span>
<span><span class="co">#&gt;   ..$ HARVEST    : num [1:2] 1989 4414</span></span>
<span><span class="co">#&gt;  $ nstates     : Named num [1:381] 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:381] "GTH2B-550" "NOD1" "Ots_100884-287" "Ots_101554-407" ...</span></span>
<span><span class="co">#&gt;  $ nalleles    : Named num [1:380] 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:380] "GTH2B-550" "NOD1" "Ots_100884-287" "Ots_101554-407" ...</span></span>
<span><span class="co">#&gt;  $ C           : int 9</span></span>
<span><span class="co">#&gt;  $ groups      :'data.frame':    43 obs. of  1 variable:</span></span>
<span><span class="co">#&gt;   ..$ D1: int [1:43] 1 1 1 1 2 2 2 2 2 3 ...</span></span>
<span><span class="co">#&gt;  $ group_names :'data.frame':    4 obs. of  1 variable:</span></span>
<span><span class="co">#&gt;   ..$ D1: chr [1:4] "Koyukuk" "Tanana" "UpperUS" "Canada"</span></span>
<span><span class="co">#&gt;  $ age_class   : Named int [1:9] 7 7 1 2 3 4 5 6 7</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr [1:9] "11" "12" "13" "21" ...</span></span>
<span><span class="co">#&gt;  $ age_classes : chr [1:7] "13" "21" "22" "23" ...</span></span>
<span><span class="co">#&gt;  $ wildpops    : chr [1:43] "KHENS01" "KHENS07.KHENS15" "KSFKOY03" "KMFKOY10.KMFKOY11.KMFKOY12.KMFKOY13" ...</span></span>
<span><span class="co">#&gt;  $ hatcheries  : chr(0) </span></span>
<span><span class="co">#&gt;  $ districts   : Named chr "1"</span></span>
<span><span class="co">#&gt;   ..- attr(*, "names")= chr "1"</span></span>
<span><span class="co">#&gt;  $ subdistricts:List of 1</span></span>
<span><span class="co">#&gt;   ..$ 1: Named int 1</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "names")= chr "1"</span></span>
<span><span class="co">#&gt;  $ stat_weeks  : int [1:2] 1 2</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-magma">Running MAGMA<a class="anchor" aria-label="anchor" href="#running-magma"></a>
</h3>
<p>Use <code><a href="../reference/magmatize_mdl.html">magmatize_mdl()</a></code> function to run the model. You’ll
need to assign the output as an object as shown in the code below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">freak_out</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/magmatize_mdl.html">magmatize_mdl</a></span><span class="op">(</span>dat_in <span class="op">=</span> <span class="va">yomamafat</span>,</span>
<span>                nreps <span class="op">=</span> <span class="fl">50</span>, nburn <span class="op">=</span> <span class="fl">25</span>, thin <span class="op">=</span> <span class="fl">1</span>, nchains <span class="op">=</span> <span class="fl">2</span>, nadapt <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                keep_burn <span class="op">=</span> <span class="cn">TRUE</span>, age_prior <span class="op">=</span> <span class="st">"zero_out"</span>,</span>
<span>                cond_gsi <span class="op">=</span> <span class="cn">TRUE</span>, file <span class="op">=</span> <span class="cn">NULL</span>, seed <span class="op">=</span> <span class="cn">NULL</span>, iden_output <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Running model... and if oppotunity doesn't knock, build Lofting!</span></span>
<span><span class="co">#&gt; Time difference of 2.572572 secs</span></span>
<span><span class="co">#&gt; 2024-10-09 19:37:37.900791</span></span></code></pre></div>
<p>Burn-ins are excluded in the summary calculations even if a user
choose to keep the burn-in output. <code>nadapt</code>,
<code>keep_burn</code>, <code>flat_age_prior</code>,
<code>cond_gsi</code>, <code>file</code> and <code>seed</code> use the
default values if not specified by the user.</p>
<p><code>cond_gsi</code> sets the option to run MAGMA model in the
conditional GSI or fully Bayesian algorithm. <code>nadapt</code> allows
for a “warm-up” run in conditional GSI mode before running the model in
fully Bayesian mode.</p>
<p>User can tinker with the <code>age_prior</code> for the MAGMA model.
<code>age_prior = "flat"</code> sets equal weights across the age
groups. <code>age_prior = "weak_flat"</code> also sets equal weights
across the age groups but with a smaller value. Specifying
<code>age_prior = "zero_out"</code> concentrates the prior weights on
the major age groups and force the undetected age groups to (near)
zero.</p>
<p><code>seed</code> allows for setting a random seed for the
pseudo-random number generator, so the output can be reproduced exactly.
Just pick a number and make a note for future reference.</p>
<p><code>iden_output</code> specifies whether or not to have the trace
history for individual group membership assignments included in the raw
model output. Default is FALSE (don’t include).</p>
<p>The raw output of MAGMA is a multi-layered list of MCMC chains. Each
chain contains a tibble with age classes
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>×</mo><annotation encoding="application/x-tex">\times</annotation></semantics></math>
iterations
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>×</mo><annotation encoding="application/x-tex">\times</annotation></semantics></math>
# of districts
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>×</mo><annotation encoding="application/x-tex">\times</annotation></semantics></math>
# of subdistricts
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>×</mo><annotation encoding="application/x-tex">\times</annotation></semantics></math>
# of weeks) as rows and populations as columns. I call the output in
“raw” format because it has not been summarized. The output also
contains the specifications for running the model (iterations, burn-ins…
all that good stuff) and the optional trace history of individual group
membership assignments.</p>
<p><code>file</code> specify the file path to save the output of
individual MCMC chain in <code>Fst</code> format at the end of the model
run. For example, if you ran MAGMA in five MCMC chains, there will be
five <code>Fst</code> files saved in the designated location.</p>
</div>
<div class="section level3">
<h3 id="summarizing-output">Summarizing Output<a class="anchor" aria-label="anchor" href="#summarizing-output"></a>
</h3>
<p>The MAGMA output is summarized using <code><a href="../reference/magmatize_summ.html">magmatize_summ()</a></code>
function. The function needs the raw MAGMA output and the input data
objects. The raw output can be read in as an <em>R</em> object or as
<code>Fst</code> files (if saved during model run). To read in the model
output as <code>Fst</code> files, use the argument <code>fst_file</code>
to specify the location of the saved files.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">magma_summ</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/magmatize_summ.html">magmatize_summ</a></span><span class="op">(</span>which_dist <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 ma_out <span class="op">=</span> <span class="va">freak_out</span>,</span>
<span>                 ma_dat <span class="op">=</span> <span class="va">yomamafat</span>,</span>
<span>                 summ_level <span class="op">=</span> <span class="st">"district"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Preparing output (patience grasshopper...)</span></span>
<span><span class="co">#&gt; Time difference of 0.5508969 secs</span></span>
<span><span class="co">#&gt; 2024-10-09 19:37:38.546584</span></span></code></pre></div>
<p>For big fisheries like TBR in SEAK, output can be too large for our
work laptops to process. The summary process can be ran one district at
a time (using argument <code>which_dist</code>) to manage memory
use.</p>
<p>MAGMA estimates age/stock composition at the lowest stratum level
(i.e., statistical weeks), but sums up the lower strata (weighted by
harvest numbers) to provide summaries at the district or subdistrict
level by setting <code>summ_level = "district"</code> or
<code>"subdistrict"</code>, respectively. Summaries are provided in
forms of stock proportions and age-by-stock compositions.</p>
<p>For the stock proportions at the district level, estimates of each
statistical week are summed up across subdistricts within each district.
There is also a summary that summed up all estimates of subdistrict
within each district.</p>
<div class="float">
<img src="pop_dist.PNG" style="width:100.0%" alt="pop_summ and pop_summ_all at the district level: Summary scheme of population proportions at the district level. Oragne arrows show how different starta are summed."><div class="figcaption">
<code>pop_summ</code> and
<code>pop_summ_all</code> at the district level: Summary scheme of
population proportions at the district level. Oragne arrows show how
different starta are summed.</div>
</div>
<p>For the age-by-stock composition at the district level, estimates of
age class proportions are summarized for each reporting group within
each district.</p>
<div class="float">
<img src="age_dist.PNG" style="width:100.0%" alt="age_summ at the district level: Summary scheme of age-by-stock composition at the district level."><div class="figcaption">
<code>age_summ</code> at the district level:
Summary scheme of age-by-stock composition at the district level.</div>
</div>
<p>For the stock proportions at the subdistrict level, estimates of each
statistical week are summarized for each subdistrict and each district.
There is also a summary that summed up all estimated proportions of
statistical weeks within each subdistrict for each district.</p>
<div class="float">
<img src="pop_subdist.PNG" style="width:100.0%" alt="pop_summ and pop_summ_all at the subdistrict level: Summary scheme of population proportions at the subdistrict level. Oragne arrows show how different starta are summed."><div class="figcaption">
<code>pop_summ</code> and
<code>pop_summ_all</code> at the subdistrict level: Summary scheme of
population proportions at the subdistrict level. Oragne arrows show how
different starta are summed.</div>
</div>
<p>For the age-by-stock composition at the subdistrict level, estimates
of age class proportions are summarized for each reporting group within
each subdistrict.</p>
<div class="float">
<img src="age_subdist.PNG" style="width:100.0%" alt="age_summ at the subdistrict level: Summary scheme of age-by-stock composition at the subdistrict level."><div class="figcaption">
<code>age_summ</code> at the subdistrict level:
Summary scheme of age-by-stock composition at the subdistrict
level.</div>
</div>
<p>The summarized MAGMA output is organized as list items. The items
with <code>_summ</code> are the summary table with posterior means,
median, CrI’s and convergence diagnostics (Gelman-Rubin and effective
size). At the district level, <code>age_summ</code> item provides
summaries of age-by-stock composition within a district.
<code>pop_summ_all</code> and <code>pop_summ</code> provide summaries of
reporting group proportions for a district and summing across
statistical weeks within a district, respectively. At the subdistrict
level, <code>age_summ</code> item provides summaries of age-by-stock
composition within each subdistrict. <code>pop_summ_all</code> and
<code>pop_summ</code> provide summaries of reporting group proportions
for each subdistrict and for each statistical week within a subdistrict,
respectively.</p>
<p>An example of the summary table:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">magma_summ</span><span class="op">$</span><span class="va">age_summ</span></span>
<span><span class="co">#&gt; $D1_Koyukuk</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">#&gt;   group  age       mean    median       sd     ci.05     ci.95    p0    GR n_eff</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Koyuk… 13    2.81<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 4.13<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 9.84<span style="color: #949494;">e</span><span style="color: #BB0000;">-  5</span> 1.07<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0.08 0.987  50  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Koyuk… 21    2.53<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.32<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 3.51<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.41<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> 1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0.06 1.21   50  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Koyuk… 22    6.53<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 5.40<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 4.25<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.38<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 1.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    0.992  29.0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Koyuk… 23    5.20<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 5.19<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 1.33<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 3.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 7.23<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    1.00   72.2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Koyuk… 31    3.40<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 3.37<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 1.25<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 1.42<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 5.25<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    0.980 159. </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Koyuk… 32    2.16<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.06<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 7.24<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> 6.11<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0.06 0.993  50  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> Koyuk… other 1.58<span style="color: #949494;">e</span><span style="color: #BB0000;">-65</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 1.12<span style="color: #949494;">e</span><span style="color: #BB0000;">-64</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 1.12<span style="color: #949494;">e</span><span style="color: #BB0000;">-130</span>  1    0.981   0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D1_Tanana</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">#&gt;   group  age        mean    median        sd     ci.05     ci.95    p0    GR</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Tanana 13    3.63<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 8.64<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0    0.999</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Tanana 21    1.70<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 1.72<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 5.40<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 9.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.52<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    0.985</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Tanana 22    3.77<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 3.82<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 8.42<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.20<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 4.95<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    1.07 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Tanana 23    3.91<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 3.77<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 9.25<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.61<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 5.42<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    1.04 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Tanana 31    1.15<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 6.82<span style="color: #949494;">e</span><span style="color: #BB0000;">-  3</span> 1.28<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.63<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> 3.61<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0.02 1.14 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Tanana 32    1.38<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 8.95<span style="color: #949494;">e</span><span style="color: #BB0000;">-  3</span> 1.67<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 6.12<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> 5.05<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0.02 1.01 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> Tanana other 2.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-101</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 1.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-100</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 9.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-184</span>  1    0.982</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: n_eff &lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D1_UpperUS</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">#&gt;   group   age       mean    median      sd     ci.05     ci.95    p0    GR n_eff</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> UpperUS 13    0.026<span style="text-decoration: underline;">1</span>   1.60<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 0.039<span style="text-decoration: underline;">4</span>  8.40<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> 7.76<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0.04 0.998  50  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> UpperUS 21    0.028<span style="text-decoration: underline;">8</span>   2.09<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 0.029<span style="text-decoration: underline;">1</span>  8.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-  4</span> 8.30<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0    0.984  50  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> UpperUS 22    0.324    3.15<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 0.097<span style="text-decoration: underline;">2</span>  1.90<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 4.96<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    1.03   83.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> UpperUS 23    0.178    1.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 0.067<span style="text-decoration: underline;">8</span>  7.17<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    1.01   50.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> UpperUS 31    0.027<span style="text-decoration: underline;">7</span>   1.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 0.034<span style="text-decoration: underline;">5</span>  1.44<span style="color: #949494;">e</span><span style="color: #BB0000;">-  3</span> 8.86<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span>  0    1.12   50  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> UpperUS 32    0.415    4.11<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 0.099<span style="text-decoration: underline;">0</span>  2.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 5.75<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span>  0    1.13   41.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> UpperUS other 0.000<span style="text-decoration: underline;">202</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.001<span style="text-decoration: underline;">42</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 4.23<span style="color: #949494;">e</span><span style="color: #BB0000;">-134</span>  0.98 1.22   25  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D1_Canada</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 10</span></span></span>
<span><span class="co">#&gt;   group  age         mean    median      sd     ci.05    ci.95    p0    GR n_eff</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Canada 13       5.50<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 4.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.72<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.89<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 9.97<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span>     0 1.24     50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Canada 21       1.17<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 1.16<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 4.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 6.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 1.86<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span>     0 1.02     50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Canada 22       3.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 5.45<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 2.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 4.49<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span>     0 0.984    50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Canada 23       3.61<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.58<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 4.94<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 3.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-  1</span> 4.42<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span>     0 1.18     50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Canada 31       4.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 4.30<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.91<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 7.73<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span>     0 1.00     50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Canada 32       5.03<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 4.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 2.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.91<span style="color: #949494;">e</span><span style="color: #BB0000;">-  2</span> 1.05<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span>     0 1.01     50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> Canada other    7.05<span style="color: #949494;">e</span><span style="color: #BB0000;">-7</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 4.98<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> 6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 1.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-44</span>     1 0.982    25</span></span></code></pre></div>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>
(or <code>n_eff</code> in the summary table) represents an estimate of
independent sample size in the posterior sample. A large
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>
is considered better than a small
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>.
Some says you would need at least blah blah number to properly estimate
the credible intervals, but there is no “official” number to go by. My
experience tells me to look at
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>
together with other diagnostics. Sometimes you may see GR (Gelman-Rubin
statistic, or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mover><mi>R</mi><mo accent="true">̂</mo></mover><annotation encoding="application/x-tex">\hat R</annotation></semantics></math>)
passes the test but
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>
is small. You may want to investigate the trace plot for that particular
output. Also, when the simulation iterations are small, the results for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mrow><mi>e</mi><mi>f</mi><mi>f</mi></mrow></msub><annotation encoding="application/x-tex">N_{eff}</annotation></semantics></math>
can get wacky.</p>
<p>The items with “<code>_prop</code>” are the posterior
samples/simulations (i.e., trace history) for age or population
proportions in a data frame. The data frame contains output from all
MCMC chains stacked together. In this example, I run two chains with 50
iterations, burn-ins of 25, and no thinning. Because I chose to keep the
burn-ins, the results are 50 rows of output in each chain. Stacking the
two chains we end up with 100 rows of posterior samples.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">magma_summ</span><span class="op">$</span><span class="va">age_prop</span></span>
<span><span class="co">#&gt; $D1_Koyukuk</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 9</span></span></span>
<span><span class="co">#&gt;      itr chain     other     `13`    `21`    `22`  `23`  `31`      `32`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     1 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.000<span style="text-decoration: underline;">118</span> 0.088<span style="text-decoration: underline;">7</span>  0.075<span style="text-decoration: underline;">0</span>  0.424 0.382 0.030<span style="text-decoration: underline;">1</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     2 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.035<span style="text-decoration: underline;">4</span>   0.128   0.042<span style="text-decoration: underline;">6</span>  0.505 0.289 0.000<span style="text-decoration: underline;">389</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     3 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.045<span style="text-decoration: underline;">5</span>   0.048<span style="text-decoration: underline;">2</span>  0.105   0.624 0.141 0.035<span style="text-decoration: underline;">5</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     4 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.016<span style="text-decoration: underline;">1</span>   0.003<span style="text-decoration: underline;">62</span> 0.009<span style="text-decoration: underline;">91</span> 0.778 0.160 0.032<span style="text-decoration: underline;">2</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     5 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.008<span style="text-decoration: underline;">44</span>  0.029<span style="text-decoration: underline;">2</span>  0.023<span style="text-decoration: underline;">4</span>  0.490 0.449 0.000<span style="text-decoration: underline;">123</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     6 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.001<span style="text-decoration: underline;">20</span>  0.026<span style="text-decoration: underline;">1</span>  0.051<span style="text-decoration: underline;">2</span>  0.762 0.152 0.006<span style="text-decoration: underline;">74</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     7 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.008<span style="text-decoration: underline;">61</span>  0.001<span style="text-decoration: underline;">36</span> 0.042<span style="text-decoration: underline;">4</span>  0.521 0.426 0.000<span style="text-decoration: underline;">025</span>6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     8 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.001<span style="text-decoration: underline;">98</span>  0.035<span style="text-decoration: underline;">8</span>  0.135   0.624 0.203 0.000<span style="text-decoration: underline;">097</span>4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     9 ch1   9.41<span style="color: #949494;">e</span><span style="color: #BB0000;">-179</span> 0.190    0.005<span style="text-decoration: underline;">94</span> 0.059<span style="text-decoration: underline;">5</span>  0.487 0.253 0.004<span style="text-decoration: underline;">94</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    10 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.002<span style="text-decoration: underline;">61</span>  0.008<span style="text-decoration: underline;">44</span> 0.103   0.458 0.429 0.000<span style="text-decoration: underline;">293</span> </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D1_Tanana</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 9</span></span></span>
<span><span class="co">#&gt;      itr chain     other   `13`   `21`  `22`  `23`   `31`    `32`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     1 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.056<span style="text-decoration: underline;">7</span> 0.165  0.291 0.363 0.048<span style="text-decoration: underline;">2</span> 0.075<span style="text-decoration: underline;">2</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     2 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.020<span style="text-decoration: underline;">9</span> 0.145  0.271 0.473 0.075<span style="text-decoration: underline;">1</span> 0.015<span style="text-decoration: underline;">1</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     3 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.013<span style="text-decoration: underline;">0</span> 0.202  0.366 0.317 0.058<span style="text-decoration: underline;">4</span> 0.044<span style="text-decoration: underline;">2</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     4 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.045<span style="text-decoration: underline;">2</span> 0.261  0.288 0.348 0.030<span style="text-decoration: underline;">7</span> 0.027<span style="text-decoration: underline;">1</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     5 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.058<span style="text-decoration: underline;">5</span> 0.229  0.304 0.390 0.015<span style="text-decoration: underline;">7</span> 0.002<span style="text-decoration: underline;">87</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     6 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.060<span style="text-decoration: underline;">8</span> 0.194  0.277 0.369 0.096<span style="text-decoration: underline;">4</span> 0.002<span style="text-decoration: underline;">53</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     7 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.100  0.186  0.227 0.410 0.036<span style="text-decoration: underline;">7</span> 0.039<span style="text-decoration: underline;">6</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     8 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.010<span style="text-decoration: underline;">8</span> 0.252  0.281 0.350 0.091<span style="text-decoration: underline;">8</span> 0.014<span style="text-decoration: underline;">1</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     9 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.015<span style="text-decoration: underline;">9</span> 0.200  0.309 0.311 0.163  0.001<span style="text-decoration: underline;">88</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    10 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.018<span style="text-decoration: underline;">8</span> 0.090<span style="text-decoration: underline;">4</span> 0.270 0.587 0.019<span style="text-decoration: underline;">5</span> 0.014<span style="text-decoration: underline;">3</span> </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D1_UpperUS</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 9</span></span></span>
<span><span class="co">#&gt;      itr chain     other     `13`    `21`  `22`  `23`    `31`  `32`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     1 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.011<span style="text-decoration: underline;">9</span>   0.080<span style="text-decoration: underline;">4</span>  0.146 0.216 0.013<span style="text-decoration: underline;">9</span>  0.531</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     2 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.000<span style="text-decoration: underline;">790</span> 0.028<span style="text-decoration: underline;">5</span>  0.345 0.185 0.040<span style="text-decoration: underline;">9</span>  0.400</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     3 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.077<span style="text-decoration: underline;">0</span>   0.007<span style="text-decoration: underline;">02</span> 0.333 0.125 0.007<span style="text-decoration: underline;">33</span> 0.451</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     4 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.039<span style="text-decoration: underline;">2</span>   0.067<span style="text-decoration: underline;">6</span>  0.164 0.184 0.080<span style="text-decoration: underline;">3</span>  0.464</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     5 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.023<span style="text-decoration: underline;">3</span>   0.045<span style="text-decoration: underline;">8</span>  0.198 0.261 0.026<span style="text-decoration: underline;">6</span>  0.445</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     6 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.014<span style="text-decoration: underline;">0</span>   0.033<span style="text-decoration: underline;">9</span>  0.225 0.113 0.040<span style="text-decoration: underline;">3</span>  0.573</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     7 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.015<span style="text-decoration: underline;">8</span>   0.046<span style="text-decoration: underline;">6</span>  0.342 0.129 0.035<span style="text-decoration: underline;">4</span>  0.430</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     8 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.016<span style="text-decoration: underline;">1</span>   0.047<span style="text-decoration: underline;">3</span>  0.514 0.117 0.001<span style="text-decoration: underline;">39</span> 0.305</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     9 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.013<span style="text-decoration: underline;">0</span>   0.003<span style="text-decoration: underline;">78</span> 0.412 0.177 0.012<span style="text-decoration: underline;">2</span>  0.382</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    10 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.002<span style="text-decoration: underline;">34</span>  0.016<span style="text-decoration: underline;">5</span>  0.266 0.220 0.015<span style="text-decoration: underline;">1</span>  0.481</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D1_Canada</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 9</span></span></span>
<span><span class="co">#&gt;      itr chain     other   `13`   `21`  `22`  `23`   `31`   `32`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>     1 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.119  0.109  0.342 0.255 0.030<span style="text-decoration: underline;">3</span> 0.144 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>     2 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.045<span style="text-decoration: underline;">5</span> 0.089<span style="text-decoration: underline;">3</span> 0.481 0.304 0.027<span style="text-decoration: underline;">9</span> 0.052<span style="text-decoration: underline;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>     3 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.107  0.094<span style="text-decoration: underline;">3</span> 0.345 0.366 0.036<span style="text-decoration: underline;">4</span> 0.051<span style="text-decoration: underline;">9</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>     4 ch1   9.77<span style="color: #949494;">e</span><span style="color: #BB0000;">- 66</span> 0.073<span style="text-decoration: underline;">3</span> 0.069<span style="text-decoration: underline;">2</span> 0.186 0.614 0.028<span style="text-decoration: underline;">4</span> 0.030<span style="text-decoration: underline;">1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>     5 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.063<span style="text-decoration: underline;">0</span> 0.071<span style="text-decoration: underline;">6</span> 0.354 0.433 0.032<span style="text-decoration: underline;">1</span> 0.046<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>     6 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.085<span style="text-decoration: underline;">2</span> 0.146  0.274 0.395 0.049<span style="text-decoration: underline;">8</span> 0.050<span style="text-decoration: underline;">6</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>     7 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.027<span style="text-decoration: underline;">9</span> 0.138  0.382 0.358 0.024<span style="text-decoration: underline;">5</span> 0.068<span style="text-decoration: underline;">9</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>     8 ch1   6.68<span style="color: #949494;">e</span><span style="color: #BB0000;">-308</span> 0.045<span style="text-decoration: underline;">3</span> 0.076<span style="text-decoration: underline;">3</span> 0.436 0.354 0.022<span style="text-decoration: underline;">4</span> 0.066<span style="text-decoration: underline;">6</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>     9 ch1   1.91<span style="color: #949494;">e</span><span style="color: #BB0000;">-202</span> 0.058<span style="text-decoration: underline;">1</span> 0.096<span style="text-decoration: underline;">2</span> 0.468 0.317 0.037<span style="text-decoration: underline;">8</span> 0.023<span style="text-decoration: underline;">2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    10 ch1   3.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-152</span> 0.029<span style="text-decoration: underline;">3</span> 0.209  0.302 0.418 0.022<span style="text-decoration: underline;">0</span> 0.020<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span></code></pre></div>
<p>Trace histories can be included as a part of the summary output
(default), or be saved as <code>Fst</code> files in a specified
location. If one decided to save the trace histories as
<code>Fst</code>, they will not be included in the summary. To save the
trace histories, use the arghument <code>save_trace</code> to specify a
location. A folder for the saved files will be automatically created.
Each file is labelled based on the type, statistical
district/subdistrict/week, and/or reporting group. For example,
<code>ap_d2Alaska.fst</code> is the trace history for age proportions of
district 2 for the Alaska reporting group. <code>p_d1s2w5.fst</code> is
the trace history for group proportions of district 1, subdistrict 2,
and week 5.</p>
<div class="section level4">
<h4 id="trace-plot">Trace Plot<a class="anchor" aria-label="anchor" href="#trace-plot"></a>
</h4>
<p>The posterior samples trace can be used to make trace plots with your
own code or with function <code><a href="../reference/magmatize_tr_plot.html">magmatize_tr_plot()</a></code>. If using
<code><a href="../reference/magmatize_tr_plot.html">magmatize_tr_plot()</a></code>, you need to specify the amount of
burn-ins and thinning you had when you ran the model. If you forget, you
can find them in your raw magma output (e.g.,
<code>magma_out$specs</code>). <code><a href="../reference/magmatize_tr_plot.html">magmatize_tr_plot()</a></code> outputs
one data frame (e.g. sampling period) at a time. The burn-in portion of
the output is shaded in red.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/magmatize_tr_plot.html">magmatize_tr_plot</a></span><span class="op">(</span><span class="va">magma_summ</span><span class="op">$</span><span class="va">age_prop</span><span class="op">$</span><span class="va">D1_Koyukuk</span>, nburn <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="magma-vignette_files/figure-html/trace-plot-1.png" alt="Trace plots for age composition." width="100%"><p class="caption">
Trace plots for age composition.
</p>
</div>
<p>Note that one can read in the saved trace histories to make trace
plots. For example, trace history for age proportions of the Koyukuk
reporting group can be plotted using
<code>magmatize_tr_plot(fst::read_fst(paste0(wd, "/trace_district/ap_d1Koyukuk.fst")), nburn = 25)</code>.</p>
</div>
<div class="section level4">
<h4 id="individual-group-membership-assignment">Individual (group membership) assignment<a class="anchor" aria-label="anchor" href="#individual-group-membership-assignment"></a>
</h4>
<p><code><a href="../reference/magmatize_indiv.html">magmatize_indiv()</a></code> summarizes the posterior means of
group membership for each individual in the metadata. The probability
output can be organized in populations or combined into reporting groups
(for single districts only).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">magma_indiv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/magmatize_indiv.html">magmatize_indiv</a></span><span class="op">(</span>ma_out <span class="op">=</span> <span class="va">freak_out</span>, ma_dat <span class="op">=</span> <span class="va">yomamafat</span>, out_repunit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Combining populations using reporting groups of district 1.</span></span>
<span></span>
<span><span class="va">magma_indiv</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 149 × 5</span></span></span>
<span><span class="co">#&gt;    indiv           Koyukuk Tanana UpperUS Canada</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> FAKE_KBEAV97_22    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> FAKE_KBEAV97_31    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> FAKE_KBEAV97_46    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> FAKE_KBEAV97_60    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> FAKE_KBEAV97_64    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> FAKE_KBEAV97_73    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> FAKE_KBEAV97_81    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> FAKE_KCHAN02_2     0.01   0.05    0.94   0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> FAKE_KCHAN04_1     0      0       0.91   0.09</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> FAKE_KCHAN04_16    0      0       1      0   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 139 more rows</span></span></span></code></pre></div>
<p>The probabilities of group memberships of each individual are based
on genotype if available. Otherwise, the posteriors are based on priors
or known identities, which provide no additional useful information on
the group memberships. The first column of the group membership
probability summary provides the unique id of each individual in the
data set, which allows users to filter individuals by name.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="background-and-methods">Background and Methods<a class="anchor" aria-label="anchor" href="#background-and-methods"></a>
</h2>
<p>We will first introduce Pella-Masuda model in this section because it
is the backbone of MAGMA. Later in the section, we will discuss how
MAGMA is developed by extending the Pella-Masuda model.</p>
<div class="section level3">
<h3 id="pella-masuda-model">Pella-Masuda Model<a class="anchor" aria-label="anchor" href="#pella-masuda-model"></a>
</h3>
<!-- $$\newcommand{\mx}[1]{\mathbf{#1}}$$ -->
<p>For a population comprised of multiple distinct groups, genetic stock
identification (GSI) is used to estimate the group membership of each
individual based on its genetic make-up (i.e., allele frequencies). The
GSI model also estimates the overall group proportions based on the
number of individuals assigned to each group. In the fishery context,
genetic data of the individuals is called the <em>mixture</em> because
it consists of multi-locus allele frequencies of individual fish
collected from a mixed-stock fishery.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐱</mi><annotation encoding="application/x-tex">\mathbf x</annotation></semantics></math>
denotes the mixture. In this document, a bold-font letter represents a
<em>number set</em>, or a collection of distinct elements. For example,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐱</mi><annotation encoding="application/x-tex">\mathbf x</annotation></semantics></math>
is a set that contains individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>
elements. And
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">x_{m,l,j}</annotation></semantics></math>
is the count of allele
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>
in locus
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>
for individual fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>m</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>M</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">m \in \{1,2,...,M\}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>l</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>L</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">l \in \{1,2,...,L\}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>j</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>J</mi><mi>l</mi></msub><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">j \in \{1,2,...,J_l\}</annotation></semantics></math>
depends on locus
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>.</p>
<p>Genetic data of the populations is called the <em>baseline</em>
because it consists allele frequencies of various reference populations
collected at their spawning locations. Researchers select sampling
locations to best represent demographic production and genetic diversity
of populations in an area.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf y</annotation></semantics></math>
denotes the baseline sample.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><annotation encoding="application/x-tex">y_{k,l,j}</annotation></semantics></math>
is the count of allele
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>j</mi><annotation encoding="application/x-tex">j</annotation></semantics></math>
in locus
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>l</mi><annotation encoding="application/x-tex">l</annotation></semantics></math>
for a sample of size
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi></mrow></msub><annotation encoding="application/x-tex">n_{k,l}</annotation></semantics></math>
collected from baseline population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>K</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">k \in \{1,2,...,K\}</annotation></semantics></math>.</p>
<p>For both mixture and baseline, it is assumed that allele counts in
each locus follow a multinomial distribution<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;What is a multinomial distribution? They are the counts
of multiple categories based on their corresponding probabilities. For
example, we randomly toss 10 Swedish Fish to three kids. Each time we
toss a fish, the probability of kid #1 catching it is 0.3, kid #2, 0.3,
and kid #3, 0.4. The numbers of fish caught by the three kids are
multinomially distributed with probabilities = {0.3, 0.3, 0.4} and a
total size of 10.&lt;/p&gt;"><sup>3</sup></a>. Using a made-up
example, in a baseline, there are two alleles in locus 1 for population
2. Counts for the two alleles are
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>y</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>y</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">y_{1,2,1},  y_{1,2,2}</annotation></semantics></math>,
and they follow a multinomial distribution with parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">q_{1,2,1}, q_{1,2,2}</annotation></semantics></math>
and size
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>n</mi><mrow><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><annotation encoding="application/x-tex">n_{2,1}</annotation></semantics></math>.
Note that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">q_{1,2,1}, q_{1,2,2}</annotation></semantics></math>
are the relative frequencies of the two alleles in locus 1 for
population 2. In a Bayesian framework, we need to specify prior
distributions for parameters; therefore, we place a Dirichlet<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;What is a Dirichlet distribution? Using the same example
for multinomial distribution, the fish catching probabilities of all
three kids are Dirichlet-distributed with fish counts of all three kids
as parameters. It is worth noting that the parameters do not have to be
integers as long as they are &amp;gt; 0.&lt;/p&gt;"><sup>4</sup></a> prior
distribution on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">q_{1,2,1}, q_{1,2,2}</annotation></semantics></math>
with hyperparameters<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;A hyperparameter is a parameter of a prior
distribution.&lt;/p&gt;"><sup>5</sup></a>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msub></mrow><annotation encoding="application/x-tex">\beta_{1,1}, \beta_{1,2}</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>1</mn></mrow></msub><mo>=</mo><msub><mi>β</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn></mrow></msub><mo>=</mo><mn>1</mn><mi>/</mi><mn>2</mn></mrow><annotation encoding="application/x-tex">\beta_{1,1} = \beta_{1,2} = 1/2</annotation></semantics></math>
based on the number of alleles for locus 1.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
represents
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>1</mn></mrow></msub><annotation encoding="application/x-tex">q_{1,2,1}</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>q</mi><mrow><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>2</mn></mrow></msub><annotation encoding="application/x-tex">q_{1,2,2}</annotation></semantics></math>,
together with allele frequencies of other loci and other populations. As
you can see,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf y</annotation></semantics></math>
have the same dimension because each relative frequency corresponds to
an allele count. In the model, allele frequencies of baseline
populations,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>,
determine population proportions. And population proportions is used to
determine the identities of individual fish. Individual identities are
then tallied and summarized to update baseline allele frequencies.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐲</mi><annotation encoding="application/x-tex">\mathbf y</annotation></semantics></math>
can be expressed as follows:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐲</mi><mi>k</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐧</mi><mi>k</mi></msub><mo>,</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf y_k \sim Mult(\mathbf n_k, \mathbf q_k)</annotation></semantics></math></p>
<p>Prior distribution for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐪</mi><mi>k</mi></msub><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf q_k \sim Dirich(\mathbf \beta)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛃</mi><mo>=</mo><mn>1</mn><mi>/</mi><msub><mi>J</mi><mi>l</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf \beta = 1/J_l</annotation></semantics></math>
is a commonly used prior for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>.</p>
<p>As mentioned, for mixture, allele counts in each locus of individual
fish follows a multinomial distributions. Distribution of allele counts
is related to the allele frequencies of the baseline population which
the individual came from. However, the identity of the individual fish
is unknown so it needs to be estimated. Here we let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐳</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf z_m</annotation></semantics></math>
represent the population identify for the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math><sup>th</sup>
mixture individual.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐳</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf z_m</annotation></semantics></math>
is composed of 0’s and an 1 with a length
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>K</mi><annotation encoding="application/x-tex">K</annotation></semantics></math>
(e.g. number of baseline populations).
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 1</annotation></semantics></math>
if individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belongs to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 0</annotation></semantics></math>
otherwise. In a made-up example,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mn>100</mn></msub><mo>=</mo><mo stretchy="false" form="prefix">{</mo><mn>0</mn><mo>,</mo><mn>0</mn><mo>,</mo><mn>1</mn><mo>,</mo><mn>0</mn><mo>,</mo><mn>0</mn><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">\mathbf z_{100} = \{0, 0, 1, 0, 0\}</annotation></semantics></math>
means that there are only five populations, and individual fish #100
comes from population 3.</p>
<p>We place a multinomial distribution on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>K</mi></mrow></msub></mrow><annotation encoding="application/x-tex">z_{m,1}, z_{m,2}, ..., z_{m,K}</annotation></semantics></math>
with size 1 and probabilities equal to population proportions
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>1</mn></msub><mo>,</mo><msub><mi>p</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>p</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">p_1, p_2, ..., p_K</annotation></semantics></math>.
We specify a Dirichlet prior distribution on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>1</mn></msub><mo>,</mo><msub><mi>p</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>p</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">p_1, p_2, ..., p_K</annotation></semantics></math>
with hyperparameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mn>1</mn></msub><mo>,</mo><msub><mi>α</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>α</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">\alpha_1, \alpha_2, ..., \alpha_K</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mn>1</mn></msub><mo>=</mo><msub><mi>α</mi><mn>2</mn></msub><mo>=</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>=</mo><msub><mi>α</mi><mi>K</mi></msub><mo>=</mo><mn>1</mn><mi>/</mi><mi>K</mi></mrow><annotation encoding="application/x-tex">\alpha_1 = \alpha_2 = ... = \alpha_K = 1/K</annotation></semantics></math>.
We express
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐳</mi><annotation encoding="application/x-tex">\mathbf z</annotation></semantics></math>
as follows:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>𝟏</mn><mo>,</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z_m \sim Mult(\mathbf 1, \mathbf p)</annotation></semantics></math></p>
<p>A commonly used prior for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf p \sim Dirich(\mathbf \alpha)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛂</mi><mo>=</mo><mn>1</mn><mi>/</mi><mi>K</mi></mrow><annotation encoding="application/x-tex">\mathbf \alpha = 1/K</annotation></semantics></math>.</p>
<p>As mentioned, for mixture sample, allele counts in each locus of
individual fish follows a multinomial distributions. The parameters are
allele frequencies of the corresponding baseline population with size
the numbers of ploidy for each respective locus. Remember that
population identity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 1</annotation></semantics></math>
if individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belongs to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">z_{m,k} = 0</annotation></semantics></math>
otherwise. When multiplying population identities,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>K</mi></mrow></msub></mrow><annotation encoding="application/x-tex">z_{m,1}, z_{m,2}, ..., z_{m,K}</annotation></semantics></math>,
and allele frequencies of baseline populations,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐪</mi><mn>1</mn></msub><mo>,</mo><msub><mi>𝐪</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>𝐪</mi><mi>K</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf q_1, \mathbf q_2, ..., \mathbf q_K</annotation></semantics></math>,
only allele frequencies of baseline population which individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to would remain while the rest goes to zero.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐱</mi><annotation encoding="application/x-tex">\mathbf x</annotation></semantics></math>
is expressed below.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐱</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>,</mo><msub><mi>𝐳</mi><mi>m</mi></msub><mo>∘</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf x_m \sim Mult(\mathbf{ploidy}, \mathbf z_m \circ \mathbf q)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>=</mo><mi>p</mi><mi>l</mi><mi>o</mi><mi>i</mi><mi>d</mi><msub><mi>y</mi><mn>1</mn></msub><mo>,</mo><mi>p</mi><mi>l</mi><mi>o</mi><mi>i</mi><mi>d</mi><msub><mi>y</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>p</mi><mi>l</mi><mi>o</mi><mi>i</mi><mi>d</mi><msub><mi>y</mi><mi>L</mi></msub></mrow><annotation encoding="application/x-tex">\mathbf{ploidy} = ploidy_1, ploidy_2, ..., ploidy_L</annotation></semantics></math>
denotes ploidy for each locus. We use
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mo>∘</mo><annotation encoding="application/x-tex">\circ</annotation></semantics></math>
to denote the element-wise product.</p>
<p>Moran and Anderson (2018) implement a genetic mixture analysis as a
<em>R</em> package, <em>rubias</em>. Their program has been widely used
by researchers around the world, including here at the GCL.
<em>rubias</em> utilizes a model structure called the conditional GSI
model, that is modified from the Pella-Masuda model. The main difference
between the two models is that, in the conditional model,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
is integrated out of the distribution of mixture sample,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐱</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf x_m</annotation></semantics></math>.
That is, baseline allele frequencies are not being updated in the model.
The result of that,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐱</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf x_m</annotation></semantics></math>
takes a form of a compound Dirichlet-multinomial distribution (Johnson
at el., 1997):</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐱</mi><mi>m</mi></msub><mo>∼</mo><mi>C</mi><mi>D</mi><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>,</mo><msub><mi>𝐳</mi><mi>m</mi></msub><mo>∘</mo><mi>𝐯</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf x_m \sim CDM(\mathbf{ploidy}, \mathbf z_m \circ \mathbf v)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐯</mi><annotation encoding="application/x-tex">\mathbf v</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛃</mi><mo>+</mo><mi>𝐲</mi></mrow><annotation encoding="application/x-tex">\mathbf \beta + \mathbf y</annotation></semantics></math>.
We are not going to attempt proving the theory behind the conditional
GSI model in this document (details can be found in Moran and Anderson,
2018). But since
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
has been integrated out of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐱</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf x_m</annotation></semantics></math>,
the process for estimating parameters is simpler and more streamlined.
We implemented a modified version of the conditional GSI in the updated
edition of MAGMA.</p>
</div>
<div class="section level3">
<h3 id="mark-and-age-inclusion">Mark and Age Inclusion<a class="anchor" aria-label="anchor" href="#mark-and-age-inclusion"></a>
</h3>
<p>MAGMA is basically Pella-Masuda model with extension to include
otolith marks and aged individual fish. In Pella-Masuda model, each fish
belongs to a wild population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>K</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">k \in \{1,2,...,K\}</annotation></semantics></math>.
And their identity is estimated based on genotype. In the extended
scenario, hatchery populations are added to the mixture and can be
identified by their otolith markings. The identities of hatchery fish
can be traced back completely to the origin
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>k</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mi>K</mi><mo>+</mo><mn>1</mn><mo>,</mo><mi>K</mi><mo>+</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>K</mi><mo>+</mo><mi>H</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">k \in \{K+1, K+2, ..., K+H\}</annotation></semantics></math>.</p>
<p>With the addition of otolith marking, the entire mixture sample of
size
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>M</mi><annotation encoding="application/x-tex">M</annotation></semantics></math>
is now comprised of three components: 1) the number of wild fish that
are genotyped
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">M^{(1)}</annotation></semantics></math>;
2) the number of wild fish that are not genotyped
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">M^{(2)}</annotation></semantics></math>;
and 3) the number of otolith-marked fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">M^{(3)}</annotation></semantics></math>.
Note that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi><mo>=</mo><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>+</mo><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>+</mo><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></mrow><annotation encoding="application/x-tex">M = M^{(1)} + M^{(2)} + M^{(3)}</annotation></semantics></math>.</p>
<p>Population identities are also partitioned into three components.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐳</mi><annotation encoding="application/x-tex">\mathbf z</annotation></semantics></math>
is now
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf z^{(1)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf z^{(2)}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf z^{(3)}</annotation></semantics></math>,
each corresponding to the respective sample-component. Compartmentalized
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>i</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\mathbf z^{(i)}_m</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>i</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mn>3</mn><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">i \in \{1,2,3\}</annotation></semantics></math>,
still follow a multinomial distribution with size 1 and parameter
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>
as described previously. However, with the addition of hatchery
populations,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>i</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\mathbf z^{(i)}_m</annotation></semantics></math>
and parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>
are now of length
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow><annotation encoding="application/x-tex">K+H</annotation></semantics></math>.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>p</mi><mn>1</mn></msub><mo>,</mo><msub><mi>p</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>p</mi><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msub></mrow><annotation encoding="application/x-tex">p_1, p_2, ..., p_{K+H}</annotation></semantics></math>
have a Dirichlet distribution with hyperparameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>α</mi><mn>1</mn></msub><mo>,</mo><msub><mi>α</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>α</mi><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msub><mo>=</mo><mn>1</mn><mi>/</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>K</mi><mo>+</mo><mi>H</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\alpha_1, \alpha_2, ..., \alpha_{K+H} = 1/(K+H)</annotation></semantics></math>.
We express
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>i</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf z^{(i)}</annotation></semantics></math>
and prior for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>
as follows:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>i</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>𝟏</mn><mo>,</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z^{(i)}_m \sim Mult(\mathbf 1, \mathbf p)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf p \sim Dirich(\mathbf \alpha)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛂</mi><mo>=</mo><mn>1</mn><mi>/</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>K</mi><mo>+</mo><mi>H</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf \alpha = 1/(K+H)</annotation></semantics></math></p>
<p>Allele counts are only available from individual fish that are
genotyped; hence, genetic information is now compartmentalized to
component 1 of the mixture sample:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>,</mo><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>⋅</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf x^{(1)}_m \sim Mult(\mathbf{ploidy}, \mathbf z^{(1)}_m \cdot \mathbf q)</annotation></semantics></math></p>
<p>It is similar for the conditional GSI model:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∼</mo><mi>C</mi><mi>D</mi><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mrow><mi>𝐩</mi><mi>𝐥</mi><mi>𝐨</mi><mi>𝐢</mi><mi>𝐝</mi><mi>𝐲</mi></mrow><mo>,</mo><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>⋅</mo><mi>𝐯</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf x^{(1)}_m \sim CDM(\mathbf{ploidy}, \mathbf z^{(1)}_m \cdot \mathbf v)</annotation></semantics></math></p>
<p>No genetic baseline samples are required for the hatchery populations
so that the genetic baseline
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>y</mi><annotation encoding="application/x-tex">y</annotation></semantics></math>
is unchanged from the Pella-Masuda model; however, no age-class baseline
is available for any population. As described earlier, some fish in the
mixture sample are aged and some are not. However, in this document we
will pretend that all fish are aged so that the notation would be less
headache-inducing. The fundamental concept would still be the same when
not all fish were aged, only with more complicated notations.</p>
<p>Age class is identified as
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>c</mi><mo>∈</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>C</mi><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">c \in \{1,2,...,C\}</annotation></semantics></math>.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐚</mi><annotation encoding="application/x-tex">\mathbf a</annotation></semantics></math>
denotes the age identities of mixture fish. Let
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐚</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf a_m</annotation></semantics></math>
represent the age identify for the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math><sup>th</sup>
mixture individual.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐚</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf a_m</annotation></semantics></math>
are also partitioned into three subsets,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐚</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf a^{(1)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐚</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf a^{(2)}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐚</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf a^{(3)}</annotation></semantics></math>,
according to the sample-components. However, it is not necessary to
compartmentalize
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐚</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf a_m</annotation></semantics></math>
for the most part in the model. Age identity and population identity
have a similar structure, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐚</mi><mi>m</mi></msub><annotation encoding="application/x-tex">\mathbf a_m</annotation></semantics></math>
is also composed of 0’s and an 1 but with a length
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>C</mi><annotation encoding="application/x-tex">C</annotation></semantics></math>.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>c</mi></mrow></msub><annotation encoding="application/x-tex">a_{m,c}</annotation></semantics></math>
is the age identity for the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math><sup>th</sup>
mixture individual in the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math><sup>th</sup>
age class, where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>c</mi></mrow></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">a_{m,c} = 1</annotation></semantics></math>
if individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
has age class
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>c</mi></mrow></msub><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">a_{m,c} = 0</annotation></semantics></math>
otherwise. For example, if there were three age classes and fish #6 was
age 3, then
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐚</mi><mn>6</mn></msub><mo>=</mo><mo stretchy="false" form="prefix">{</mo><mn>0</mn><mo>,</mo><mn>0</mn><mo>,</mo><mn>1</mn><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">\mathbf a_6 = \{0, 0, 1\}</annotation></semantics></math>.</p>
<p>We place a multinomial distribution on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>C</mi></mrow></msub></mrow><annotation encoding="application/x-tex">a_{m,1}, a_{m,2}, ..., a_{m,C}</annotation></semantics></math>
with size 1 and probabilities equal to age-class frequencies
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐳</mi><mi>m</mi></msub><mo>⋅</mo><mi>𝛑</mi></mrow><annotation encoding="application/x-tex">\mathbf z_m \cdot \mathbf \pi</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛑</mi><annotation encoding="application/x-tex">\mathbf \pi</annotation></semantics></math>
denotes age-class frequencies within each population 1 through
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow><annotation encoding="application/x-tex">K+H</annotation></semantics></math>.
You can picture
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛑</mi><annotation encoding="application/x-tex">\mathbf \pi</annotation></semantics></math>
as a matrix with
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow><annotation encoding="application/x-tex">K+H</annotation></semantics></math>
populations of rows and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>C</mi><annotation encoding="application/x-tex">C</annotation></semantics></math>
age classes of columns. Therefore, when multiplying population
identities,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>K</mi></mrow></msub></mrow><annotation encoding="application/x-tex">z_{m,1}, z_{m,2}, ..., z_{m,K}</annotation></semantics></math>,
and age-class frequencies,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝛑</mi><mrow><mn>1</mn><mo>,</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>C</mi><mo stretchy="false" form="postfix">}</mo></mrow></msub><mo>,</mo><msub><mi>𝛑</mi><mrow><mn>2</mn><mo>,</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>C</mi><mo stretchy="false" form="postfix">}</mo></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>𝛑</mi><mrow><mi>K</mi><mo>+</mo><mi>H</mi><mo>,</mo><mo stretchy="false" form="prefix">{</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>C</mi><mo stretchy="false" form="postfix">}</mo></mrow></msub></mrow><annotation encoding="application/x-tex">\mathbf \pi_{1,\{1,2,...,C\}}, \mathbf \pi_{2,\{1,2,...,C\}}, ..., \mathbf \pi_{K+H,\{1,2,...,C\}}</annotation></semantics></math>,
only age-class frequencies within the population which individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to would remain while the rest goes to zero.</p>
<p>We specify a Dirichlet prior on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mn>1</mn></mrow></msub><mo>,</mo><msub><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mn>2</mn></mrow></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>C</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\pi_{k,1}, \pi_{k,2}, ..., \pi_{k,C}</annotation></semantics></math>
with hyperparameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>γ</mi><mn>1</mn></msub><mo>,</mo><msub><mi>γ</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>γ</mi><mi>C</mi></msub><mo>=</mo><mn>1</mn><mi>/</mi><mi>C</mi></mrow><annotation encoding="application/x-tex">\gamma_1, \gamma_2, ..., \gamma_C = 1/C</annotation></semantics></math>.
We express
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐚</mi><annotation encoding="application/x-tex">\mathbf a</annotation></semantics></math>
and prior for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛑</mi><annotation encoding="application/x-tex">\mathbf \pi</annotation></semantics></math>
as follows:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐚</mi><mi>m</mi></msub><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>𝟏</mn><mo>,</mo><msub><mi>𝐳</mi><mi>m</mi></msub><mo>⋅</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf a_m \sim Mult(\mathbf 1, \mathbf z_m \cdot \mathbf \pi)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛑</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf \pi \sim Dirich(\mathbf \gamma)</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛄</mi><mo>=</mo><mn>1</mn><mi>/</mi><mi>C</mi></mrow><annotation encoding="application/x-tex">\mathbf \gamma = 1/C</annotation></semantics></math></p>
<p>Estimation of MAGMA parameters requires deriving the conditional
distributions for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐚</mi><mo>,</mo><mi>𝛂</mi><mo>,</mo><mi>𝛃</mi><mo>,</mo><mi>𝛄</mi></mrow><annotation encoding="application/x-tex">\mathbf p, \mathbf q, \mathbf z^{(1)}, \mathbf z^{(2)}, \mathbf\pi|\mathbf x, \mathbf y, \mathbf z^{(3)}, \mathbf a, \mathbf\alpha, \mathbf\beta, \mathbf\gamma</annotation></semantics></math>.
In the next section, we will introduce the concepts and an algorithm to
sample the posterior distribution.</p>
<div class="float">
<img src="magma_dag.png" style="width:100.0%" alt="MAGMA DAG: A simplified directed acyclic graph of MAGMA model. Ovals represent variables in the model. Shaded boxes represent observed variables. Diamond shapes represent the prior distributions. The large rectangles represent replication over individuals and/or loci. See main text for full explanation of the model."><div class="figcaption">MAGMA DAG: A simplified directed acyclic graph
of MAGMA model. Ovals represent variables in the model. Shaded boxes
represent observed variables. Diamond shapes represent the prior
distributions. The large rectangles represent replication over
individuals and/or loci. See main text for full explanation of the
model.</div>
</div>
</div>
<div class="section level3">
<h3 id="gibbs-sampler">Gibbs Sampler<a class="anchor" aria-label="anchor" href="#gibbs-sampler"></a>
</h3>
<p>Gibbs sampler is a type of Markov chain Monte Carlo (MCMC) methods
that sequentially sample parameter values from a Markov chain. With
enough sampling, the Markov chain will eventually converge to the joint
posterior distribution of the parameters. The most appealing quality of
Gibbs sampler is its reduction of a multivariate problem (such as
Pella-Masuda and MAGMA models) to a series of more manageable
lower-dimensional problems. A full description of Gibbs sampler and MCMC
methods is beyond the scope of this document; however, further
information can be found in numerous resources devoting to Bayesian data
analysis (see Carlin &amp; Louis, 2009; Robert &amp; Casella, 2010;
Gelman et al., 2014)</p>
<p>To illustrate, suppose we would like to determine the joint posterior
distribution of interest,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \theta|\mathbf y)</annotation></semantics></math>,
where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝛉</mi><mo>=</mo><mo stretchy="false" form="prefix">{</mo><msub><mi>θ</mi><mn>1</mn></msub><mo>,</mo><msub><mi>θ</mi><mn>2</mn></msub><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msub><mi>θ</mi><mi>K</mi></msub><mo stretchy="false" form="postfix">}</mo></mrow><annotation encoding="application/x-tex">\mathbf \theta = \{\theta_1, \theta_2,..., \theta_K\}</annotation></semantics></math>.
Most likely the multivariate
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \theta|\mathbf y)</annotation></semantics></math>
would be too complicated to sample from. However, if we can figure out
how to break up the joint posterior distribution into individual full
conditional distributions<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;The distribution of a parameter conditioned on the data
and other parameters in the model.&lt;/p&gt;"><sup>6</sup></a>, each parameter in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛉</mi><annotation encoding="application/x-tex">\mathbf \theta</annotation></semantics></math>
can be sampled one by one sequentially using a Gibbs sampler algorithm.
The process begins with an arbitrary set of starting values
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mi>K</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></mrow><annotation encoding="application/x-tex">\theta^{(0)}_2, \theta^{(0)}_3,..., \theta^{(0)}_K</annotation></semantics></math>
and proceeds as follows:</p>
<p>For
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>t</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">t = 1,2,...,T</annotation></semantics></math>,
repeat</p>
<ol style="list-style-type: decimal">
<li><p>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>θ</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\theta^{(t)}_1</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>θ</mi><mn>1</mn></msub><mo stretchy="false" form="prefix">|</mo><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\theta_1|\theta^{(t-1)}_2, \theta^{(t-1)}_3,..., \theta^{(t-1)}_k, \mathbf y)</annotation></semantics></math></p></li>
<li>
<p>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\theta^{(t)}_2</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>θ</mi><mn>2</mn></msub><mo stretchy="false" form="prefix">|</mo><msubsup><mi>θ</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>3</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\theta_2|\theta^{(t)}_1, \theta^{(t-1)}_3,..., \theta^{(t-1)}_k, \mathbf y)</annotation></semantics></math></p>
<p>⋮</p>
</li>
</ol>
<ol start="11" style="list-style-type: lower-alpha">
<li>Draw
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>θ</mi><mi>k</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><annotation encoding="application/x-tex">\theta^{(t)}_k</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>θ</mi><mi>k</mi></msub><mo stretchy="false" form="prefix">|</mo><msubsup><mi>θ</mi><mn>1</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msubsup><mi>θ</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><msubsup><mi>θ</mi><mrow><mi>k</mi><mo>−</mo><mn>1</mn></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝐲</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\theta_k|\theta^{(t)}_1, \theta^{(t)}_2,..., \theta^{(t)}_{k-1}, \mathbf y)</annotation></semantics></math>
</li>
</ol>
<p>This would work best if the full conditionals are some known
distributions that we can easily sample from (although it’s not
required). In our case with MAGMA model, we rely on two main concepts,
the Bayes theorem and conjugacy, to do the trick. Briefly, for
estimating parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛉</mi><annotation encoding="application/x-tex">\mathbf\theta</annotation></semantics></math>
from data
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐃</mi><annotation encoding="application/x-tex">\mathbf D</annotation></semantics></math>,
according to Bayes Rule,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mfrac><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">p(\mathbf\theta|\mathbf D) = \displaystyle \frac{p(\mathbf D|\mathbf\theta)p(\mathbf\theta)}{p(\mathbf D)}</annotation></semantics></math>.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf\theta|\mathbf D)</annotation></semantics></math>
is the joint posterior distribution for parameters
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛉</mi><annotation encoding="application/x-tex">\mathbf\theta</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D|\mathbf\theta)</annotation></semantics></math>
is the likelihood of observing the data given the parameters,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf\theta)</annotation></semantics></math>
is the prior distribution of the parameters, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D)</annotation></semantics></math>
is the constant marginal distribution of the data.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D)</annotation></semantics></math>
is often mathematically difficult to obtain; however, because
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf D)</annotation></semantics></math>
is a constant number, we can ignore it by reducing the posterior
distribution to
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐃</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛉</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf\theta|\mathbf D) \propto p(\mathbf D|\mathbf\theta)p(\mathbf\theta)</annotation></semantics></math>.</p>
<p>So, how does Bayes Rule help us estimating parameters in MAGMA model?
First, the joint posterior distribution has to be split up into smaller
pieces. That is, we separate the joint posterior into likelihood of the
data and priors for the parameters:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><mi>𝐚</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛂</mi><mo>,</mo><mi>𝛃</mi><mo>,</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p, \mathbf q, \mathbf z^{(1)}, \mathbf z^{(2)}, \mathbf\pi|\mathbf x, \mathbf y, \mathbf a, \mathbf z^{(3)}, \mathbf\alpha, \mathbf\beta, \mathbf\gamma)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\propto p(\mathbf x|\mathbf z^{(1)}, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf a|\mathbf z,\mathbf\pi) p(\mathbf z^{(3)}|\mathbf p) \cdot p(\mathbf p|\mathbf\alpha) p(\mathbf q|\mathbf\beta) p(\mathbf z^{(1)}|\mathbf p) p(\mathbf z^{(2)}|\mathbf p) p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">= p(\mathbf x|\mathbf z^{(1)}, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf a|\mathbf z,\mathbf\pi) \cdot p(\mathbf p|\mathbf\alpha) p(\mathbf q|\mathbf\beta) p(\mathbf z|\mathbf p) p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math></p>
<p>With some re-arrangements and hand-waving, we arrive at the
conditional distributions for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛑</mi><annotation encoding="application/x-tex">\mathbf\pi</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z^{(1)}, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf a|\mathbf z,\mathbf\pi) \cdot p(\mathbf p|\mathbf\alpha) p(\mathbf q|\mathbf\beta) p(\mathbf z|\mathbf p) p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">= p(\mathbf x|\mathbf z^{(1)}, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf q|\mathbf\beta) \cdot p(\mathbf z|\mathbf p) p(\mathbf p|\mathbf\alpha) \cdot p(\mathbf a|\mathbf z,\mathbf\pi) p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo>,</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\propto p(\mathbf x,\mathbf y,\mathbf z^{(1)}|\mathbf q) p(\mathbf q|\mathbf\beta) \cdot p(\mathbf z|\mathbf p) p(\mathbf p|\mathbf\alpha) \cdot p(\mathbf a,\mathbf z|\mathbf\pi) p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐚</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\propto p(\mathbf q|\mathbf x,\mathbf y,\mathbf z^{(1)},\mathbf\beta) \cdot p(\mathbf p|\mathbf z,\mathbf\alpha) \cdot p(\mathbf \pi|\mathbf a,\mathbf z,\mathbf\gamma)</annotation></semantics></math></p>
<p>Next, we take advantage of a mathematical property called
<em>conjugacy</em> to help us determine the full conditional
distributions. Based on this property, the posterior distribution
follows the same parametric form as the prior distribution when prior is
a <em>conjugate family</em> for the likelihood. For example, if the
likelihood of data is binomial distribution and the prior of parameter
is beta distribution, then the posterior is also beta distribution
because beta is a conjugate family for binomial. There are many
conjugate families, and Dirichlet and multinomial are another
example.</p>
<p>Utilizing conjugacy property, we will determine each of the
conditional distributions for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐩</mi><annotation encoding="application/x-tex">\mathbf p</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝛑</mi><annotation encoding="application/x-tex">\mathbf\pi</annotation></semantics></math>.</p>
<div class="section level4">
<h4 id="conditional-distribution-pqx-y-z1-beta">Conditional distribution p(q|x, y, z<sup>(1)</sup>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>)<a class="anchor" aria-label="anchor" href="#conditional-distribution-pqx-y-z1-beta"></a>
</h4>
<p>We determine that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf x,\mathbf y,\mathbf z^{(1)},\mathbf\beta)</annotation></semantics></math>
is Dirichlet-distributed because Dirichlet prior
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf\beta)</annotation></semantics></math>
is a conjugate family for the multinomial likelihoods
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z^{(1)}, \mathbf q)</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf y|\mathbf q)</annotation></semantics></math>.
To determine the exact parameterization for the posterior distribution,
we need to derive the prior and likelihoods first.</p>
<p>Likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z^{(1)}, \mathbf q)</annotation></semantics></math>
can be derived in two steps. The first step we conditioned the
likelihood on
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf z^{(1)}</annotation></semantics></math>
so that</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msup><mrow><mo stretchy="true" form="prefix">[</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></msup></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z^{(1)}, \mathbf q) \propto \displaystyle \prod^{M^{(1)}}_{m=1} \prod^{K}_{k=1} [f(\mathbf x_m|\mathbf q_k)]^{z^{(1)}_{m,k}}</annotation></semantics></math>,</p>
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k)</annotation></semantics></math>
is the relative frequency of multi-locus genotype for individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>.
In the next step, we derive
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k)</annotation></semantics></math>:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup></mrow><annotation encoding="application/x-tex">f(\mathbf x_m|\mathbf q_k) \propto \displaystyle \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{x_{m,l,j}}_{k,l,j}</annotation></semantics></math></p>
<p>Then we combine the two,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msup><mrow><mo stretchy="true" form="prefix">[</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐱</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></msup></mrow><annotation encoding="application/x-tex">p(\mathbf x|\mathbf z^{(1)}, \mathbf q) \propto \displaystyle \prod^{M^{(1)}}_{m=1} \prod^{K}_{k=1} [f(\mathbf x_m|\mathbf q_k)]^{z^{(1)}_{m,k}}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><mrow><mo stretchy="true" form="prefix">[</mo><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></mrow></msubsup><mo stretchy="true" form="postfix">]</mo></mrow></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{M^{(1)}}_{m=1} \prod^{K}_{k=1} [\displaystyle \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{x_{m,l,j} \cdot z^{(1)}_{m,k}}_{k,l,j}]</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M^{(1)}}_{m=1} (x_{m,l,j} \cdot z^{(1)}_{m,k})}_{k,l,j}</annotation></semantics></math></p>
<p>Deriving likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf y|\mathbf q)</annotation></semantics></math>
is more straightforward. It is the product of relative frequency of
multi-locus genotype for each population:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup></mrow><annotation encoding="application/x-tex">p(\mathbf y|\mathbf q) \propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{y_{k,l,j}}_{k,l,j}</annotation></semantics></math></p>
<p>And
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>q</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(q|\mathbf\beta)</annotation></semantics></math>
is Dirichlet prior distribution. Its probability density has a kernel<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;Without involving too much math, &lt;em&gt;kernel&lt;/em&gt; here
refers to what is left over after factoring out the constant from a
probability density function (PDF). For example, The PDF of Dirichlet
distribution is
&lt;math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mfrac&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;mrow&gt;&lt;mi&gt;B&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy="true" form="prefix"&gt;(&lt;/mo&gt;&lt;mi&gt;𝛂&lt;/mi&gt;&lt;mo stretchy="true" form="postfix"&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;/mfrac&gt;&lt;msubsup&gt;&lt;mo&gt;∏&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;/msubsup&gt;&lt;msubsup&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;α&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mo&gt;−&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;\frac{1}{B(\mathbf \alpha)} \displaystyle \prod^K_{i=1} x^{\alpha_i - 1}_i&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;.
Beta function
&lt;math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mi&gt;B&lt;/mi&gt;&lt;mrow&gt;&lt;mo stretchy="true" form="prefix"&gt;(&lt;/mo&gt;&lt;mi&gt;𝛂&lt;/mi&gt;&lt;mo stretchy="true" form="postfix"&gt;)&lt;/mo&gt;&lt;/mrow&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;B(\mathbf \alpha)&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;
can be factored out, and
&lt;math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;msubsup&gt;&lt;mo&gt;∏&lt;/mo&gt;&lt;mrow&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mo&gt;=&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;mi&gt;K&lt;/mi&gt;&lt;/msubsup&gt;&lt;msubsup&gt;&lt;mi&gt;x&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;mrow&gt;&lt;msub&gt;&lt;mi&gt;α&lt;/mi&gt;&lt;mi&gt;i&lt;/mi&gt;&lt;/msub&gt;&lt;mo&gt;−&lt;/mo&gt;&lt;mn&gt;1&lt;/mn&gt;&lt;/mrow&gt;&lt;/msubsup&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;\displaystyle \prod^K_{i=1} x^{\alpha_i - 1}_i&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;
is the kernel.&lt;/p&gt;'><sup>7</sup></a> of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mi>𝛃</mi><mo>−</mo><mn>1</mn></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{\mathbf \beta - 1}</annotation></semantics></math>.
We can express the likelihood as</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf\beta) \propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math>.</p>
<p>Put all the likelihoods together,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐱</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐲</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐪</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf x,\mathbf y,\mathbf z^{(1)},\mathbf\beta) \propto p(\mathbf x|\mathbf z^{(1)}, \mathbf q) p(\mathbf y|\mathbf q) p(\mathbf q|\mathbf\beta)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M^{(1)}}_{m=1} (x_{m,l,j} \cdot z^{(1)}_{m,k})}_{k,l,j} \cdot \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{y_{k,l,j}}_{k,l,j} \cdot \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">= \displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M^{(1)}}_{m=1} (x_{m,l,j} \cdot z^{(1)}_{m,k}) + y_{k,l,j} + \beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math></p>
<p>It is <em>elementary</em> for anybody to recognize that
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>l</mi><mo>=</mo><mn>1</mn></mrow><mi>L</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>j</mi><mo>=</mo><mn>1</mn></mrow><msub><mi>J</mi><mi>l</mi></msub></msubsup><msubsup><mi>q</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>⋅</mo><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\displaystyle \prod^{K}_{k=1} \prod^{L}_{l=1} \prod^{J_l}_{j=1} q^{\sum^{M^{(1)}}_{m=1} (x_{m,l,j} \cdot z^{(1)}_{m,k}) + y_{k,l,j} + \beta_{l,j} - 1}_{k,l,j}</annotation></semantics></math>
is the kernel for Dirichlet distribution. Hence,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝐪</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi></mrow></msub><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛃</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><msup><mi>M</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup></msubsup><msub><mi>x</mi><mrow><mi>m</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><msubsup><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>+</mo><msub><mi>y</mi><mrow><mi>k</mi><mo>,</mo><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo>+</mo><msub><mi>β</mi><mrow><mi>l</mi><mo>,</mo><mi>j</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf q_{k,l}|\mathbf x,\mathbf y,\mathbf z^{(1)},\mathbf\beta \sim Dirich(\displaystyle \sum^{M^{(1)}}_{m=1} x_{m,l,j} z^{(1)}_{m,k} + y_{k,l,j} + \beta_{l,j})</annotation></semantics></math></p>
</div>
<div class="section level4">
<h4 id="conditional-distribution-ppz-alpha">Conditional distribution p(p|z,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>α</mi><annotation encoding="application/x-tex">\alpha</annotation></semantics></math>)<a class="anchor" aria-label="anchor" href="#conditional-distribution-ppz-alpha"></a>
</h4>
<p>Using the same logic as previously,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf z,\mathbf\alpha)</annotation></semantics></math>
is also Dirichlet-distributed due to a Dirichlet prior
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf\alpha)</annotation></semantics></math>
and a multinomial likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf z|\mathbf p)</annotation></semantics></math>.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐳</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf z,\mathbf\alpha) \propto p(\mathbf z|\mathbf p) p(\mathbf p|\mathbf\alpha)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mi>p</mi><mi>k</mi><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mi>p</mi><mi>k</mi><mrow><msub><mi>α</mi><mi>k</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{M}_{m=1} \prod^{K+H}_{k=1} p^{z_{m,k}}_k \cdot \prod^{K+H}_{k=1} p^{\alpha_k - 1}_k</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mi>p</mi><mi>k</mi><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>α</mi><mi>k</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{K+H}_{k=1} p^{\sum^M_{m=1}z_{m,k} + \alpha_k - 1}_k</annotation></semantics></math></p>
<p>Once again, we recognize it as the kernel for Dirichlet
distribution:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛂</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>α</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf p|\mathbf z,\mathbf\alpha \sim Dirich(\displaystyle \sum^M_{m=1}z_{m,k} + \alpha_k)</annotation></semantics></math></p>
</div>
<div class="section level4">
<h4 id="conditional-distribution-ppia-z-gamma">Conditional distribution
p(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>|a,
z,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>γ</mi><annotation encoding="application/x-tex">\gamma</annotation></semantics></math>)<a class="anchor" aria-label="anchor" href="#conditional-distribution-ppia-z-gamma"></a>
</h4>
<p>Lastly,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐚</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \pi|\mathbf a,\mathbf z,\mathbf\gamma)</annotation></semantics></math>
is also Dirichlet-distributed due to a Dirichlet prior
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math>
and a multinomial likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf a|\mathbf z,\mathbf\pi)</annotation></semantics></math>.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐚</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐚</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐳</mi><mo>,</mo><mi>𝛑</mi><mo stretchy="true" form="postfix">)</mo></mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \pi|\mathbf a,\mathbf z,\mathbf\gamma) \propto p(\mathbf a|\mathbf z,\mathbf\pi) p(\mathbf \pi|\mathbf\gamma)</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msup><mrow><mo stretchy="true" form="prefix">[</mo><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐚</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msub><mi>γ</mi><mi>c</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\propto \displaystyle \prod^{M}_{m=1}\prod^{K+H}_{k=1}[h(\mathbf a_m|\mathbf\pi_k)]^{z_{m,k}} \cdot \prod^{K+H}_{k=1}\prod^{C}_{c=1}\pi^{\gamma_c - 1}_{k,c}</annotation></semantics></math>,</p>
<p>where likelihood
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐚</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>∝</mo><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msubsup></mrow><annotation encoding="application/x-tex">h(\mathbf a_m|\mathbf\pi_k) \propto \displaystyle \prod^C_{c=1}\pi^{a_{m,k}}_{k,c}</annotation></semantics></math>
is the product of relative frequency of age class for individual
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>.
Plugging in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐚</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">h(\mathbf a_m|\mathbf\pi_k)</annotation></semantics></math>,</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msup><mrow><mo stretchy="true" form="prefix">[</mo><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>𝐚</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo stretchy="true" form="postfix">]</mo></mrow><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></msup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msub><mi>γ</mi><mi>c</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">\displaystyle \prod^{M}_{m=1}\prod^{K+H}_{k=1}[h(\mathbf a_m|\mathbf\pi_k)]^{z_{m,k}} \cdot \prod^{K+H}_{k=1}\prod^{C}_{c=1}\pi^{\gamma_c - 1}_{k,c}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><msubsup><mo>∏</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><mrow><mo stretchy="true" form="prefix">[</mo><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></mrow></msubsup><mo stretchy="true" form="postfix">]</mo></mrow><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msub><mi>γ</mi><mi>c</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">= \displaystyle \prod^{M}_{m=1}\prod^{K+H}_{k=1}[\displaystyle \prod^C_{c=1}\pi^{a_{m,k}z_{m,k}}_{k,c}] \cdot \prod^{K+H}_{k=1}\prod^{C}_{c=1}\pi^{\gamma_c - 1}_{k,c}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub></mrow></msubsup><mo>⋅</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msub><mi>γ</mi><mi>c</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">= \displaystyle \prod^{K+H}_{k=1}\displaystyle \prod^C_{c=1}\pi^{\sum^M_{m=1}a_{m,k}z_{m,k}}_{k,c} \cdot \prod^{K+H}_{k=1}\prod^{C}_{c=1}\pi^{\gamma_c - 1}_{k,c}</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo>=</mo><msubsup><mo>∏</mo><mrow><mi>k</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>K</mi><mo>+</mo><mi>H</mi></mrow></msubsup><msubsup><mo>∏</mo><mrow><mi>c</mi><mo>=</mo><mn>1</mn></mrow><mi>C</mi></msubsup><msubsup><mi>π</mi><mrow><mi>k</mi><mo>,</mo><mi>c</mi></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>γ</mi><mi>c</mi></msub><mo>−</mo><mn>1</mn></mrow></msubsup></mrow><annotation encoding="application/x-tex">= \displaystyle \prod^{K+H}_{k=1}\displaystyle \prod^C_{c=1}\pi^{\sum^M_{m=1}a_{m,k}z_{m,k}+ \gamma_c - 1}_{k,c}</annotation></semantics></math></p>
<p>And we recognize it as the kernel for Dirichlet distribution:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="false" form="prefix">|</mo><mi>𝐚</mi><mo>,</mo><mi>𝐳</mi><mo>,</mo><mi>𝛄</mi><mo>∼</mo><mi>D</mi><mi>i</mi><mi>r</mi><mi>i</mi><mi>c</mi><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mo>∑</mo><mrow><mi>m</mi><mo>=</mo><mn>1</mn></mrow><mi>M</mi></msubsup><msub><mi>a</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><msub><mi>z</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>+</mo><msub><mi>γ</mi><mi>c</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf \pi_k|\mathbf a, \mathbf z, \mathbf\gamma \sim Dirich(\displaystyle \sum^M_{m=1}a_{m,k}z_{m,k} + \gamma_c)</annotation></semantics></math></p>
</div>
<div class="section level4">
<h4 id="algorithm">Algorithm<a class="anchor" aria-label="anchor" href="#algorithm"></a>
</h4>
<p>There is one more distribution to figure out before we can start our
Gibbs sampler routine (and you thought we’re all set, lol). We would
need to know how to sample
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝛑</mi><mo>,</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></mrow><annotation encoding="application/x-tex">\mathbf z^{(1,2)}_m|\mathbf p, \mathbf q, \mathbf x^{(1,2)}_m, \mathbf\pi, \mathbf a^{(1,2)}_m</annotation></semantics></math>,
the population identity for individual fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
(in components 1 and 2) given the population proportions, genotype, and
age. If the probability of fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>p</mi><mi>k</mi></msub><annotation encoding="application/x-tex">p_k</annotation></semantics></math>,
and the likelihood of observing relative frequency of genotype and age
class for fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x^{(1,2)}_m|\mathbf q_k) \cdot h(\mathbf a^{(1,2)}_m|\mathbf\pi_k)</annotation></semantics></math>,
then the probability of fish
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
belong to population
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>k</mi><annotation encoding="application/x-tex">k</annotation></semantics></math>
given the population proportions genotype, and age is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mfrac><mrow><msub><mi>p</mi><mi>k</mi></msub><mo>⋅</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><mrow><msubsup><mo>∑</mo><mrow><mi>k</mi><mi>′</mi><mo>=</mo><mn>1</mn></mrow><mi>K</mi></msubsup><msub><mi>p</mi><mrow><mi>k</mi><mi>′</mi></mrow></msub><mo>⋅</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mrow><mi>k</mi><mi>′</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mrow><mi>k</mi><mi>′</mi></mrow></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><annotation encoding="application/x-tex">\displaystyle \frac{p_k \cdot f(\mathbf x^{(1,2)}_m|\mathbf q_k) \cdot h(\mathbf a^{(1,2)}_m|\mathbf\pi_k)}{\sum^K_{k'=1}p_{k'} \cdot f(\mathbf x^{(1,2)}_m|\mathbf q_{k'}) \cdot h(\mathbf a^{(1,2)}_m|\mathbf\pi_{k'})}</annotation></semantics></math>.
The denominator should sum to one, so we only need to calculate the
numerator.
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝛑</mi><mo>,</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup></mrow><annotation encoding="application/x-tex">\mathbf z^{(1,2)}_m|\mathbf p, \mathbf q, \mathbf x^{(1,2)}_m, \mathbf\pi, \mathbf a^{(1,2)}_m</annotation></semantics></math>
has the following distribution:</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><mi>𝐩</mi><mo>,</mo><mi>𝐪</mi><mo>,</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><mi>𝛑</mi><mo>,</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msub><mrow><mi>𝐩</mi><mi mathvariant="bold">′</mi></mrow><mi>m</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf z^{(1,2)}_m|\mathbf p, \mathbf q, \mathbf x^{(1,2)}_m, \mathbf\pi, \mathbf a^{(1,2)}_m \sim Mult(1, \mathbf{p'}_m)</annotation></semantics></math>,</p>
<!-- where $p'_{m,k} = \displaystyle \frac{p_k \cdot f(\mathbf x^{(1,2)}_m|\mathbf q_k) \cdot h(\mathbf a^{(1,2)}_m|\mathbf\pi_k)}{\sum^K_{k'=1}p_{k'} \cdot f(\mathbf x^{(1,2)}_m|\mathbf q_{k'}) \cdot h(\mathbf a^{(1,2)}_m|\mathbf\pi_{k'})}$. -->
<p>where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><msub><mi>′</mi><mrow><mi>m</mi><mo>,</mo><mi>k</mi></mrow></msub><mo>=</mo><msub><mi>p</mi><mi>k</mi></msub><mo>⋅</mo><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow><mo>⋅</mo><mi>h</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝛑</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p'_{m,k} = p_k \cdot f(\mathbf x^{(1,2)}_m|\mathbf q_k) \cdot h(\mathbf a^{(1,2)}_m|\mathbf\pi_k)</annotation></semantics></math>.
We draw the initial values for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝐪</mi><mi>k</mi></msub><annotation encoding="application/x-tex">\mathbf q_k</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>𝛑</mi><mi>k</mi></msub><annotation encoding="application/x-tex">\mathbf \pi_k</annotation></semantics></math>
based on their prior distributions.</p>
<p>Once we figured out all the pieces in the Gibbs sampler, we may begin
the process with starting values for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf p^{(0)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{(0)}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝛑</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf \pi^{(0)}</annotation></semantics></math>.
If not all fish were aged,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>0</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow></msubsup><annotation encoding="application/x-tex">\mathbf a^{(1,2)(0)}_m</annotation></semantics></math>
at the initial step would contain all 0’s for those individuals without
an assigned age. Which is not a problem because age class will be
determined in the subsequent steps. We proceed as follows:</p>
<p>For
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>t</mi><mo>=</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo>,</mo><mi>.</mi><mi>.</mi><mi>.</mi><mo>,</mo><mi>T</mi></mrow><annotation encoding="application/x-tex">t = 1,2,...,T</annotation></semantics></math>,
repeat</p>
<ol style="list-style-type: decimal">
<li><p>Determine the population membership of mixture individuals,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msup><mi>𝛑</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msub><mrow><mi>𝐩</mi><mi mathvariant="bold">′</mi></mrow><mi>m</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">{{\mathbf z}^{(1,2)}_m}^{(t)}|\mathbf p^{(t-1)}, \mathbf q^{(t-1)}, \mathbf x^{(1,2)}_m, \mathbf\pi^{(t-1)}, {{\mathbf a}^{(1,2)}_m}^{(t-1)} \sim Mult(1, \mathbf {p'}_m)</annotation></semantics></math>.</p></li>
<li><p>If not all individuals were aged, determine memberships of age
classes for those with unknown age,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mi>𝐚</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo>,</mo><msup><mi>𝛑</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>∼</mo><mi>M</mi><mi>u</mi><mi>l</mi><mi>t</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><msubsup><mi>𝐳</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><msup><mi>𝛑</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo>−</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\mathbf a^{(t)}_m|\mathbf z^{(t)}_m, \mathbf \pi^{(t-1)} \sim Mult(1, \mathbf z^{(t)}_m \mathbf\pi^{(t-1)})</annotation></semantics></math>.</p></li>
<li><p>Draw updated values for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐪</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf q^{(t)}</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝐩</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf p^{(t)}</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>𝛑</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><annotation encoding="application/x-tex">\mathbf\pi^{(t)}</annotation></semantics></math>
from
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐪</mi><mo stretchy="false" form="prefix">|</mo><mi>𝐱</mi><mo>,</mo><mi>𝐲</mi><mo>,</mo><msup><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo stretchy="true" form="postfix">)</mo></mrow></msup><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛃</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf q|\mathbf x,\mathbf y,{{\mathbf z}^{(1)}}^{(t)},\mathbf\beta)</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝐩</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛂</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf p|\mathbf z^{(t)},\mathbf \alpha)</annotation></semantics></math>,
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>𝛑</mi><mo stretchy="false" form="prefix">|</mo><msup><mi>𝐚</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><msup><mi>𝐳</mi><mrow><mo stretchy="true" form="prefix">(</mo><mi>t</mi><mo stretchy="true" form="postfix">)</mo></mrow></msup><mo>,</mo><mi>𝛄</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">p(\mathbf \pi|\mathbf a^{(t)},\mathbf z^{(t)},\mathbf \gamma)</annotation></semantics></math>
respectively.</p></li>
</ol>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>T</mi><annotation encoding="application/x-tex">T</annotation></semantics></math>
should be large enough to ensure the sampler chain converges to the
posterior distribution of the parameters. Usually it takes thousands of
iterations. That completes the Gibbs sampler process. Whew.</p>
<p>Implementing the conditional GSI model only requires a slight
modification from the above algorithm. Basically,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mrow><mo stretchy="true" form="prefix">(</mo><msubsup><mi>𝐱</mi><mi>m</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>,</mo><mn>2</mn><mo stretchy="true" form="postfix">)</mo></mrow></msubsup><mo stretchy="false" form="prefix">|</mo><msub><mi>𝐪</mi><mi>k</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">f(\mathbf x^{(1,2)}_m|\mathbf q_k)</annotation></semantics></math>
would only need to be derived once in the beginning of the process, and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>𝐪</mi><annotation encoding="application/x-tex">\mathbf q</annotation></semantics></math>
would no longer need to be updated in step 3. Everything else would stay
the same.</p>
<p>we eventually realized that a purely conditional GSI algorithm would
not work for MAGMA because the baseline allele frequencies needed to be
updated so that the age frequencies would be updated as well. Instead,
we adapted an algorithm that is the hybrid of conditional and fully
Bayesian GSI. Mainly, we would run the model in the conditional GSI
algorithm with the fully Bayesian algorithm at every 10<sup>th</sup>
iteration.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Carlin, B. and T. Louis. 2009. <em>Bayesian Methods for Data
Analysis, 3rd Edition</em>. CRC Press. New York.</p>
<p>Gelman, A., J. Carlin, H. Stern, D. Dunson, A. Vehtari and D. Rubin.
<em>Bayesian Data Analysis, 3rd Edition</em>. CRC Press. New York.</p>
<p>Johnson, N.L., Kotz, S., and Balakrishnan, N. 1997. Discrete
multivariate distributions. Wiley &amp; Sons, New York.</p>
<p>Moran, B.M. and E.C. Anderson. 2018. Bayesian inference from the
conditional genetic stock identification model. <em>Canadian Journal of
Fisheries and Aquatic Sciences</em>. 76(4):551-560. <a href="https://doi.org/10.1139/cjfas-2018-0016" class="external-link uri">https://doi.org/10.1139/cjfas-2018-0016</a></p>
<p>Pella, J. and M. Masuda. 2001. Bayesian methods for analysis of stock
mixtures from genetic characters. <em>Fish. Bull.</em> 99:151–167.</p>
<p>Robert, C. and G. Casella. 2010. <em>Introducing Monte Carlo Methods
with R</em>. Springer. New York.</p>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bobby Hsu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
